### ISSUE TITLE
improved docs about creating a custom plugin

### ISSUE THREAD

**Issue by oprogramador** [eslint/eslint#12901]
I found https://eslint.org/docs/developer-guide/working-with-plugins but it doesn't contain enough info.

I know that the `create` method should return an object having `Literal` and `TemplateElement` methods. Are there any other methods I can implement? I don't know which method I should implement in order to validate comments.

**Comment by oprogramador** [comment]
maybe `LineComment` and `BlockComment` but I checked none of them is invoked

**Comment by oprogramador** [comment]
I can implement the `Program` method like in https://github.com/eslint/eslint/blob/master/lib/rules/no-warning-comments.js though it works a bit in a different way than other methods

**Comment by kaicataldo** [comment]
Have you seen [the documentation for how rules work](https://eslint.org/docs/developer-guide/working-with-rules)? If so, do you have any suggestions for how we might be able to make things clearer?

PRs to clarify documentation are always appreciated!

**Comment by kaicataldo** [comment]
Regarding accessing comments, since they're not part of the AST, we don't visit them. The best way to access all the comments can be seen [here](https://github.com/eslint/eslint/blob/master/lib/rules/capitalized-comments.js#L293-L297):

```js
Program() {
    // Get an array of all the comments in the file.
    const comments = sourceCode.getAllComments();

    // Iterate over the array and check the comments!
    comments.forEach(comment => checkComment(comment));
}
```
